mlflow/tracking/__init__.py

```diff
@@ -353,11 +352,17 @@ def _get_legacy_artifact_repo(file_store, run_info):
 
 
 def _get_git_commit(path):
+    try:
+        from git import Repo, InvalidGitRepositoryError, GitCommandNotFound
+    except ImportError, e:
+        print("Notice: failed to import git (the git executable is probably not on your PATH),"
+              " so git sha will not be available. Error: %s" % e, file=sys.stderr)
+        return None 
     try:
         if os.path.isfile(path):
             path = os.path.dirname(path)
         repo = Repo(path, search_parent_directories=True)
         commit = repo.head.commit.hexsha
         return commit
-    except (InvalidGitRepositoryError, ValueError):
+    except (InvalidGitRepositoryError, GitCommandNotFound, ValueError):
```

Comment:
NB: If the user follows the instructions provided by the error (setting `GIT_PYTHON_REFRESH=quiet`), then the import no longer throws, but use of git instead throws `GitCommandNotFound`.